tools/SourceKit/tools/sourcekitd/lib/API/Requests.cpp

```diff
@@ -2430,9 +2433,24 @@ bool SKEditorConsumer::handleSourceText(StringRef Text) {
   return true;
 }
 
-bool SKEditorConsumer::handleSerializedSyntaxTree(StringRef Text) {
-  if (syntaxTreeEnabled())
-    Dict.set(KeySerializedSyntaxTree, Text);
+bool SKEditorConsumer::handleSyntaxTree(
+    swift::syntax::SourceFileSyntax SyntaxTree,
+    std::unordered_set<unsigned> ReusedNodeIds) {
+  if (Opts.SyntaxTransferMode == SyntaxTreeTransferMode::Off)
+    return true;
```

Comment:
(I know it's currently discarded) What's the semantics of the return value? Shouldn't this be `false`?

Reply:
I assumed it was whether the function finished without error. But I think in practice it's always `true`. Maybe we should change all of these functions to return `void`.

Reply:
I will tackle this in a short follow-up PR since it's pretty unrelated to this one.